DataAccessLayer/Repository/Interfaces/IBaseRepository.cs

```diff
@@ -0,0 +1,10 @@
+﻿using System.Linq.Expressions;
+
+namespace DataAccessLayer.Repository.Interfaces;
+
+public interface IBaseRepository<TEntity> where TEntity : class
+{
+    Task<List<TEntity>> GetAllAsync();
```

Comment:
It would be better if your repository interface will include all crud operations